packages/bot/src/transformers/message.ts

```diff
@@ -410,7 +410,7 @@ export function transformMessageInteractionMetadata(bot: Bot, payload: DiscordMe
   if (props.triggeringInteractionMetadata && payload.triggering_interaction_metadata)
     metadata.triggeringInteractionMetadata = transformMessageInteractionMetadata(bot, payload.triggering_interaction_metadata)
   if (props.type) metadata.type = payload.type
-  if (props.userId) metadata.userId = bot.transformers.snowflake(payload.user_id)
+  if (props.user && payload.user) metadata.user = bot.transformers.user(bot, payload.user)
```

Comment:
Well no, but actually yes.
Right now, sometimes we pass partial objects to transformers and abused `// @ts-expect-error` as well as pray that we handled optional payload properly so these checks are needed.
Fleny has opened https://github.com/discordeno/discordeno/issues/3613 because he "started thinking that the abuse of ts-expect-error is getting out of hand" so yeah